app/services/catalog/update_order_item.rb

```diff
@@ -31,30 +29,37 @@ def find_order_item
 
     def fetch_external_url
       TopologicalInventory.call do |api_instance|
-        @service_instance_id = @task.context[:service_instance][:id]
-        service_instance = api_instance.show_service_instance(@service_instance_id)
-        raise ServiceInstanceWithoutExternalUrl if service_instance.external_url.nil?
+        service_instance = api_instance.show_service_instance(service_instance_id)
+        if service_instance.external_url.nil?
+          Rails.logger.warn("Could not find an external url on service instance (id: #{service_instance_id}) attached to task_id: #{@payload["task_id"]}")
```

Comment:
since `service_instance` is already a variable. Should we use `service_instance.id` instead of re calling the `service_instance_id` method?

Reply:
I see we're memoizing the `service_instance_id` method. That is good enough for me.